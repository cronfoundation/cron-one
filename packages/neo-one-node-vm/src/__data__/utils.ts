import { utils } from '@neo-one/client-common';
import { StorageItem } from '@neo-one/node-core';

// tslint:disable-next-line no-any
export const verifyBlockchainSnapshot = (blockchain: any) => {
  Object.entries(blockchain).forEach(([name, obj]) => {
    if (typeof obj === 'object' && obj !== null) {
      Object.entries(obj).forEach(([mockName, maybeMock]) => {
        if (maybeMock != undefined && maybeMock.mock != undefined) {
          expect(`${name}.${mockName}`).toMatchSnapshot();
          expect(maybeMock.mock.calls).toMatchSnapshot();
        }
      });
    }
  });
};

// tslint:disable-next-line no-any
export const verifyListeners = (listeners: any) => {
  // tslint:disable-next-line no-any
  Object.values(listeners).forEach((func: any) => {
    expect(func.mock.calls).toMatchSnapshot();
  });
};

export const expectItemBNEquals = (item: StorageItem, value: string) => {
  expect(utils.fromSignedBuffer(item.value).toString(10)).toEqual(value);
};

// tslint:disable-next-line:no-any readonly-array
export const badSeen = (...args: any[]) => new Set(args);
