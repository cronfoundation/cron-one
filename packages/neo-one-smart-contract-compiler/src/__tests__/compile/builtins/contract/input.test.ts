import { helpers } from '../../../../__data__';

describe('Input', () => {
  test('cannot be implemented', async () => {
    helpers.compileString(
      `
      import { Input } from '@neo-one/smart-contract';

      class MyInput implements Input {
      }
    `,
      { type: 'error' },
    );
  });
});
