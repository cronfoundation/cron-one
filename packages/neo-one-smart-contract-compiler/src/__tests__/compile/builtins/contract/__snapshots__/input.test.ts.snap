// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input cannot be implemented 1`] = `
"snippetCode.ts (4,13): Class 'MyInput' incorrectly implements interface 'Input'.
  Type 'MyInput' is missing the following properties from type 'Input': hash, index, [OpaqueTagSymbol0]

      2 |       import { Input } from '@neo-one/smart-contract';
      3 | 
    > 4 |       class MyInput implements Input {
        |             ^
      5 |       }
      6 |     
"
`;
